### DISCUSSION TITLE
Use configs and emqx envs in plugin

### DISCUSSION THREAD

**Discussion by alexa22013** [emqx/emqx#11275] [Q&A]
How can I create a configuration file for emqx plugin and access it in the main file? 
Also how can I access envs defined during cluster deployment? I was tying to get envs from _Env variable but its an empty list

Thanks in advance!

**Comment by id** [comment]
I need to check with the team.
Meanwhile, I tested that it works if you include configuration in `src/*.app.src` file, under `env`.

**Comment by id** [comment]
Another option is to get configuration from OS environment. You need to implement support for this in the plugin via `https://www.erlang.org/doc/man/os.html#getenv-1`.

**Reply by alexa22013** [reply]
can we get config from etc/plugin.config file?
